src/Features/Core/Portable/EditAndContinue/ActiveStatementFlags.cs

```diff
@@ -30,5 +30,21 @@ internal enum ActiveStatementFlags
         /// The statement IL is not in user code.
         /// </summary>
         NonUserCode = 4,
+
+        /// <summary>
+        /// Indicates that the active statement instruction belongs to the latest version of the containing method.
+        /// If not set, the containing method was updated but the active statement was not remapped yet because the thread 
+        /// has not returned to that instruction yet and was not remapped to the new version.
```

Comment:
It creates a new metadata and IL structures for it. The new version of the method is jitted the first time any thread calls it or returns to it. If a thread returns to a method (via return or throw) the execution still continues in the old method for a bit, until it can get hijacked (remapped) by the CLR runtime and moved to the new version.



---
In reply to: [162465125](https://github.com/dotnet/roslyn/pull/24316#discussion_r162465125) [](ancestors = 162465125)